src/pybind/mgr/orchestrator.py

```diff
@@ -579,37 +577,12 @@ def __init__(self, host_pattern, data_devices, db_devices=None, wal_devices=None
         #: A :class:`orchestrator.DeviceSelection`
         self.data_devices = data_devices
 
-        #: A :class:`orchestrator.DeviceSelection`
-        self.db_devices = db_devices
```

Comment:
Instead of removing these unused fields (and I agree with your point in removing them!), I'd prefer if we could make use of them in the ssh orchestrator. This way, we would have a real use case that sets the standard for other orchestrators.

I do worry about chaos when multiple orchestrators want to make use of it and every orchestrator invents a different way.

Reply:
We definitely wouldn't want them doing that.

Typically what happens is that one orchestrator will implement a new field (like db_devices) and when another wants to implement similar functionality, they just use the same field in the object.

I've just found that bloating out data structures with "aspirational" fields ahead of time is generally counterproductive. It adds cruft that often ends up just being replaced with something better suited for the purpose, once we do get around to implementing it.